pkg/app/piped/deploysource/deploysource.go

```diff
@@ -90,7 +90,7 @@ func (p *provider) Get(ctx context.Context, lw io.Writer) (*DeploySource, error)
 
 	if !p.done {
 		p.source, p.err = p.prepare(ctx, lw)
-		p.done = true
+		p.done = p.err == nil // If there is an error, we should re-prepare it next time.
```

Comment:
📝 This modification is necessary to avoid an error when executing a cancellation while executing TargetDSP.Get in another stage.

I encountered such a situation when canceling the deployment via WebUI during the SCRIPT_RUN stage execution.

<img width="1153" alt="Cursor_と_PipeCD__1__png" src="https://github.com/user-attachments/assets/ae674866-b346-4732-b798-fb5c7fc51240">

Reply:
I split the fixes to the another PR https://github.com/pipe-cd/pipecd/pull/5216